engine/src/main/java/org/hibernate/search/engine/metadata/impl/DocumentFieldMetadata.java

```diff
@@ -216,13 +226,14 @@ public String toString() {
 
 		public Builder(BackReference<TypeMetadata> sourceType,
 				BackReference<PropertyMetadata> sourceProperty,
-				String absoluteName,
+				String absoluteName, String relativeName,
```

Comment:
If feels really weird to have the `absoluteName` and the `relativeName`. Maybe it's time to introduce a `FieldPath` enclosing the prefix and the relative name and having a method building the absolute name?